examples/flutter_gallery/lib/gallery/options.dart

```diff
@@ -0,0 +1,466 @@
+  final ValueChanged<GalleryOptions> onOptionsChanged;
+
+  @override
+  Widget build(BuildContext context) {
+    return new _OptionsItem(
+      child: new Row(
+        children: <Widget>[
+          new Expanded(
+            child: new Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: <Widget>[
+                const Text('Text size'),
+                 new Text(
+                   '${options.textScaleFactor.label}',
+                   style: Theme.of(context).primaryTextTheme.body1,
+                 ),
+              ],
+            ),
+          ),
+          new PopupMenuButton<GalleryTextScaleValue>(
+            padding: const EdgeInsets.only(right: 16.0),
+            icon: const Icon(Icons.arrow_drop_down),
+            itemBuilder: (BuildContext context) {
+              return kAllGalleryTextScaleValues.map((GalleryTextScaleValue scaleValue) {
+                return new PopupMenuItem<GalleryTextScaleValue>(
+                  value: scaleValue,
+                  child: new Text(scaleValue.label),
+                );
+              }).toList();
+            },
+            onSelected: (GalleryTextScaleValue scaleValue) {
+              onOptionsChanged(
+                options.copyWith(textScaleFactor: scaleValue),
+              );
+            },
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _TextDirectionItem extends StatelessWidget {
+  const _TextDirectionItem(this.options, this.onOptionsChanged);
+
+  final GalleryOptions options;
+  final ValueChanged<GalleryOptions> onOptionsChanged;
+
+  @override
+  Widget build(BuildContext context) {
+    return new _BooleanItem(
+      'Force RTL',
+      options.textDirection == TextDirection.rtl,
+      (bool value) {
+        onOptionsChanged(
+          options.copyWith(
+            textDirection: value ? TextDirection.rtl : TextDirection.ltr,
+          ),
+        );
+      },
+    );
+  }
+}
+
+class _TimeDilationItem extends StatelessWidget {
+  const _TimeDilationItem(this.options, this.onOptionsChanged);
+
+  final GalleryOptions options;
+  final ValueChanged<GalleryOptions> onOptionsChanged;
+
+  @override
+  Widget build(BuildContext context) {
+    return new _BooleanItem(
+      'Slow motion',
+      options.timeDilation != 1.0,
+      (bool value) {
+        onOptionsChanged(
+          options.copyWith(
+            timeDilation: value ? 20.0 : 1.0,
+          ),
+        );
+      },
+    );
+  }
+}
+
+class _PlatformItem extends StatelessWidget {
+  const _PlatformItem(this.options, this.onOptionsChanged);
+
+  final GalleryOptions options;
+  final ValueChanged<GalleryOptions> onOptionsChanged;
+
+  String _platformLabel(TargetPlatform platform) {
+    switch(platform) {
+      case TargetPlatform.android:
+        return 'Android';
+      case TargetPlatform.fuchsia:
+        return 'Fuchsia';
+      case TargetPlatform.iOS:
+        return 'iOS';
+    }
+    assert(false);
```

Comment:
The return below this assert is dead code; without it flutter lint complains. As far as I can tell, this is the conventional way we've handled this in the framework's source code.

```
assert(false); return null;
```

Reply:
oh lol, I see. It's an analyzer issue.

Can't you just add a
```dart
default:
  assert(false, 'Unhandled TargetPlatform type');
```
or something?

Reply:
Sure, but I think we'd want to do that throughout the framework source code to be consistent. Otherwise it looks like this code is satisfying a unique requirement.